ClientApp/viking-vault/src/components/TopBar.tsx

```diff
@@ -1,13 +1,58 @@
 import React from 'react'
 import './styles.css'
 import logo from './images/logo.png'
+import { Link } from 'react-router-dom';
+import {variables} from "../ConstantVariables";
 
-class TopBar extends React.Component{
+const baseUrl = variables.baseUrl;
+
+interface IState {
+    isAdmin: boolean;
+}
+
+class TopBar extends React.Component<any, IState>{
+    constructor(props:any) {
+        super(props);
+        this.state = {
```

Comment:
For constructors that simply set a state, you can get rid of them and set the state as a class property
https://miro.medium.com/max/700/0*OKQ-h5NIfbiH8OQD.png